### PR TITLE
Ensure newly created or dirty and reattached views are reachable by change detection traversal

### LINKED ISSUE #52928: Views attached inside OnPush components are not checked in some circumstences
### Which @angular/* package(s) are the source of the bug?

core

### Is this a regression?

No

### Description

Currently attaching a view doesn't mark ancestor views for check. This means that there is no way to get to a new view, even if it is dirty initally.

### Please provide a link to a minimal reproduction of the bug

https://stackblitz.com/edit/onpush-dynamic-creation-qxpe1m?file=src%2Fmain.ts

### Please provide the exception or error you saw

There is no error but the newly inserted view is not change detected.

### Please provide the environment you discovered this bug in (run `ng version`)

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by atscott**
See individual commits

fixes #52928

### CODE REVIEW

File: packages/core/src/render3/util/view_utils.ts [hunk 1]
```diff
@@ -208,7 +208,7 @@ export function requiresRefreshOrTraversal(lView: LView) {
 export function updateAncestorTraversalFlagsOnAttach(lView: LView) {
   // When we attach a view that's marked `Dirty`, we should ensure that it is reached during the
   // next CD traversal so we add the `RefreshView` flag and mark ancestors accordingly.
-  if (lView[FLAGS] & LViewFlags.Dirty) {
+  if (lView[FLAGS] & LViewFlags.Dirty || (lView[FLAGS] & LViewFlags.FirstLViewPass)) {
```
Comment (alxhub) [inline packages/core/src/render3/util/view_utils.ts:211]:
```suggestion
  if (lView[FLAGS] & (LViewFlags.Dirty | LViewFlags.FirstLViewPass)) {
```

Comment (alxhub) [reply packages/core/src/render3/util/view_utils.ts:211]: Should we make `LView` just always be created with the `Dirty` flag instead?

File: packages/core/src/render3/instructions/change_detection.ts [hunk 2]
```diff
@@ -45,6 +45,12 @@ export function detectChangesInternal(lView: LView, notifyErrorHandler = true) {
   try {
     const tView = lView[TVIEW];
     const context = lView[CONTEXT];
+
+    const consumer = lView[REACTIVE_TEMPLATE_CONSUMER];
+    if (consumer) {
+      consumer.dirty = false;
+    }
+    lView[FLAGS] &= ~(LViewFlags.HasChildViewsToRefresh | LViewFlags.RefreshView);
```
Comment (alxhub) [inline packages/core/src/render3/instructions/change_detection.ts:53]: This seems like a change from a prior PR

**Review by alxhub — APPROVED** [review]

**Comment by atscott** [comment]
[global presubmit](https://fusion2.corp.google.com/presubmit/584048261/OCL:584048261:BASE:589126885:1702049887157:b09f6dd1;groups=PossiblyNewlyFailing/targets) had ~5 total failures:

* 2 failures were test setups that did not set required inputs correctly. Now that the test component actually executes, it threw errors
* 1 a11y test that appears to be a result of autofocus now working _correctly_ where it did no before
* 1 screenshot test with a small shift in pixels for a tooltip
* (the big one): 1 failure that relied on the bug being fixed. Component initialization attempted to read the value of an input where that value was set by `ngModel`. That happens in a `Promise` so the component relied on _not_ getting change detected in the current cycle. Note that this should really be `ExpressionChanged...` error anyways but `checkNoChanges` was blocked by an `OnPush` parent

The last failure is concerning and while the new behavior is still technically correct, this change can lead to actual production issues. As a result, this PR is updated to have the fix controlled by a flag that is off right now. We will roll this out when possible in g3 with the private exports to opt out _temporarily_. This flag will also be set to true by default for the v18 release.

**Comment by atscott** [comment]
This PR was merged into the repository by commit 606de5166e33affd7bd53d1ad86c11afa3cdbe21.

**Pull request closed without merging**